utils/resources.py

```diff
@@ -0,0 +1,101 @@
+import json
+import os
+import shutil
+import sublime
+import subprocess
+import threading
+
+from sublime_lib import ResourcePath
+
+
+def run_command(on_exit, on_error, popen_args):
+    """
+    Runs the given args in a subprocess.Popen, and then calls the function
+    on_exit when the subprocess completes.
+    on_exit is a callable object, and popen_args is a list/tuple of args that
+    on_error when the subprocess throws an error
+    would give to subprocess.Popen.
+    """
+    def run_in_thread(on_exit, on_error, popen_args):
+        try:
+            subprocess.check_call(popen_args, shell=sublime.platform() == 'windows')
+            on_exit()
+        except subprocess.CalledProcessError as error:
+            on_error(error)
+
+    thread = threading.Thread(target=run_in_thread, args=(on_exit, on_error, popen_args))
+    thread.start()
+    # returns immediately after the thread starts
+    return thread
+
+
+def log_and_show_message(msg, additional_logs=None):
+    print(msg, '\n', additional_logs) if additional_logs else print(msg)
+    sublime.active_window().status_message(msg)
+
+
+class ServerNpmResources(object):
```

Comment:
Sorry for more work for you but I decided to do a big refactor.

Moved `ServerResources` class into a separate file and called it more appropriately `ServerNpmResources`. My main goal was to make it easily reusable (for `LSP-vue` and maybe other packages in the future).

Reply:
> reusable  (for LSP-vue and maybe other packages in the future)

For LSP-vue and other LSP-* plugins, 
With the approach you made
I concluded that we only need to make one folder (for example named server) and place the package.json file in that folder. and that is it :)
```
> plugin.py
> server
    > package.json

PACKAGE_NAME = 'LSP-vue'
SETTINGS_FILENAME = 'LSP-vue.sublime-settings'
SERVER_DIRECTORY = 'server'
SERVER_BINARY_PATH = os.path.join(SERVER_DIRECTORY, 'node_modules', 'vue-language-server', 'bin', 'vls')

server_resources = ServerNpmResources(PACKAGE_NAME, SERVER_DIRECTORY, SERVER_BINARY_PATH)

# and later call server_resorce.initialize_resources()
```

Did I conclude something wrong?

Reply:
The only thing that I would suggest is to change some variable names :)
https://github.com/sublimelsp/LSP-eslint/commit/a4df8f3345ec3abbeba75a25844d17a29a8012ab

Otherwise this is better than before :)

Later, we would just need to move the `utils/resources.py`  to LSP so other LSP-* could use that class also.

Reply:
Yes, exactly like that for `LSP-vue`.

Reply:
Applied your change with a small tweak to restore initialization of `self._package_cache_path`